src/shared/install.c

```diff
@@ -1720,19 +1720,65 @@ static int install_info_symlink_alias(
         assert(config_path);
 
         STRV_FOREACH(s, i->aliases) {
+
                 _cleanup_free_ char *alias_path = NULL, *dst = NULL;
 
                 q = install_full_printf(i, *s, &dst);
                 if (q < 0)
                         return q;
 
+                if (unit_name_is_valid(i->name, UNIT_NAME_TEMPLATE) && !unit_name_is_valid(dst, UNIT_NAME_TEMPLATE) && !i->default_instance)
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Alias for templates without DefaultInstance must be templates.");
```

Comment:
"Alias must be ... templates" — number mismatch. Maybe "Aliases for a template without DefaultInstance= must also be templates." ?